### PR TITLE
Code review2 ck

### PR DESCRIPTION
**Pull request by ConnorKeithCK**

### CODE REVIEW

File: app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainViewModel.kt [hunk 1]
```diff
@@ -34,10 +34,6 @@ class MainViewModel : ViewModel() {
         weatherLocations = weatherService.fetchAllWeather()
     }
 
-    fun fetchAllWeather() {
```
Comment (ConnorKeithCK) [inline app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainViewModel.kt:37]: Simply just removing technical debt here, really has no impact on the project/code itself, I just noticed that this method was already defined in the same class and this particular method was never being called.

File: app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainViewModel.kt [hunk 2]
```diff
@@ -13,7 +13,7 @@ import edu.ucandroid.weathernotice.service.LocationService
 
 class MainViewModel : ViewModel() {
 
-    private lateinit var firestore : FirebaseFirestore
```
Comment (ConnorKeithCK) [inline app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainViewModel.kt:16]: Lateinit is unnecessary here, as this variable is guaranteed to be initialized via the constructor calls from FirebaseFirestore. 

https://kotlinlang.org/docs/properties.html#late-initialized-properties-and-variables

File: app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainFragment.kt [hunk 3]
```diff
@@ -51,7 +51,7 @@ class MainFragment : Fragment() {
     }
 
     lateinit var mFusedLocationProviderClient: FusedLocationProviderClient
-    val locationRequestId = 100
+    val LOCATION_REQUEST_ID = 100
```
Comment (ConnorKeithCK) [inline app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainFragment.kt:54]: Simply renaming this variable to better reflect coding practices when defining constant variables :)

https://medium.com/incwell-bootcamp/declaring-variables-in-kotlin-64d6261144ae
https://kotlinlang.org/docs/coding-conventions.html#property-names

**Comment by ConnorKeithCK** [comment]
**Analysis of the program:**

I was very impressed with the work the group has accomplished so far. The app itself compiles and functions as intended (from what I can tell from the readme, at least) and looks stellar. To be completely honest, I really struggled with finding suggestions for this code review, hence the rather nitpicky findings/comments. Overall, the code is very well written!

**Was the program available in UC Github on time?**

Yes

**Is the program documented/commented well enough for you to understand?**

The code could certainly use some more comments just to help navigate the codebase, however it also isn't that hard to follow the logic and function calls. Part of that is because they have named their function calls and classes very accurately resembling their purpose. 

**Does the program compile?**

Yes


**Rationale behind your changes.**

As mentioned above, I had a really hard time finding significant issues/suggestions in the codebase. Most of my changes are very basic and really have no impact at all on the functionality or quality of the code, merely just little tiny pieces that were missed, like changing a variable declaration and removing an unnecessary method. Overall, the code is incredibly well written. 

The one change that I did make that wasn't really "mundane" was in regards to the redundant lateinit cast to the FirebaseFirestore variable. Lateinit is unecessary there as it is guaranteed to not be null from FirebaseFirestore. I left a link in that specific commit with more information about that :)


Overall, very good job by this team!


**What I learned**

Gestures: To be completely honest, I totally thought gestures were handled by the device mostly, and just required a simple function call/event listener, like onLeftSwipe(). I was quite wrong, obviously. After looking at your codebase and the professor's videos, I learned that it is actually much much more involved than that. 

https://gyazo.com/5ceca3de7ced8f03a3ace7724beaf786

Location: My team's app is not going to be utilizing location (at least not at the moment, we might end up implementing it if we can think of a feature that requires it but nonetheless), therefore I wasn't really familiar with how location requests work, beyond the professor's videos. I was also surprised with how intricate of a process requesting and using location is. On kind of a side note, this isn't really a code suggestion, but more of a user experience suggestion. I noticed when searching for locations, you can only search by City & Country. Obviously, this will come down to the location service you are using, but as a user I would prefer to see City, State, Country or even just a disclaimer stating that is what will be returned when searching :)

`fun getLocation() {

        if (checkForLocationPermission()) {
            updateLocation()
        } else {
            askLocationPermission()
        }
    }

    fun updateLocation() {
        var locationRequest = LocationRequest()
        locationRequest.priority = LocationRequest.PRIORITY_LOW_POWER
        locationRequest.interval = 60000
        locationRequest.fastestInterval = 5000

        mFusedLocationProviderClient = FusedLocationProviderClient( context!!)


        if (ContextCompat.checkSelfPermission(
                        context!!,
                        Manifest.permission.ACCESS_FINE_LOCATION
            ) != PackageManager.PERMISSION_GRANTED && ContextCompat.checkSelfPermission(
                        context!!,
                        Manifest.permission.ACCESS_COARSE_LOCATION
             ) != PackageManager.PERMISSION_GRANTED)

        {

            return
        }
        mFusedLocationProviderClient.requestLocationUpdates(
                locationRequest, mLocationCallback,
                Looper.myLooper()
        )
    }


    var mLocationCallback = object : LocationCallback() {
        override fun onLocationResult(p0: LocationResult) {

            var location: Location = p0.lastLocation

            updateAddressUI(location)

        }
    }

    fun updateAddressUI(location: Location) {


        var addressList = ArrayList<Address>()

        var geocoder: Geocoder = Geocoder(activity!!.applicationContext, Locale.getDefault())

        addressList = geocoder.getFromLocation(
                location.latitude,
                location.longitude,
                1
        ) as ArrayList<Address>
        /** this get the full address **/
        //enterCityname.setText(addressList.get(0).getAddressLine(0))
        /** this get the city only from address **/
        enterCityname.setText(addressList[0].locality + " , " + addressList[0].countryCode )



    }


    fun checkForLocationPermission(): Boolean {

        if (ContextCompat.checkSelfPermission(
                        context!!,
                        Manifest.permission.ACCESS_FINE_LOCATION)
                == PackageManager.PERMISSION_GRANTED
        )
            return true

        return false

    }


    fun askLocationPermission() {

        requestPermissions(arrayOf(Manifest.permission.ACCESS_FINE_LOCATION),
                LOCATION_REQUEST_ID)


    }


    override fun onRequestPermissionsResult(
            requestCode: Int,
            permissions: Array<out String>,
            grantResults: IntArray
    ) {

        if (requestCode == LOCATION_REQUEST_ID) {

            if (grantResults != null && grantResults[0] == PackageManager.PERMISSION_GRANTED) {

                getLocation()
            }
        }

    }`


Inflaters: I had no idea what an inflater was before reviewing the code. I am notoriously awful with UI, so I typically try to stray away from it as long as possible, but I did some research thanks to y'alls code and know understand what exactly they are! While I don't think I will really need to use them as the business logic guy for my team, it is still nice to learn something new :)

`return inflater.inflate(R.layout.main_fragment, container, false)`

File: app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainViewModel.kt (hunk 1 above)
Comment (Hitchnt) [reply app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainViewModel.kt:37]: I believe that one of the member plan still using it for a another use in another branch. That this time I do think was a use for this in another branch that load data into a recycler view.

File: app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainViewModel.kt (hunk 2 above)
Comment (Hitchnt) [reply app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainViewModel.kt:16]: Acknowledge.  I sure I the that did this, so thank you for the correction.

File: app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainFragment.kt (hunk 3 above)
Comment (Hitchnt) [reply app/src/main/java/edu/ucandroid/weathernotice/ui/main/MainFragment.kt:54]: I think I fix this but may change back due to merging that we did going to master. Still a great catch.

**Comment by discospiff** [comment]
Good back and forth conversation with the reviewer and group... so, no further comment needed from me!
